src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -659,8 +661,27 @@ private List<ODataPathSegment> ProcessSelectTokenPath(PathSegmentToken tokenIn)
             Debug.Assert(tokenIn != null, "tokenIn != null");
 
             List<ODataPathSegment> pathSoFar = new List<ODataPathSegment>();
+
             IEdmStructuredType currentLevelType = this.edmType;
 
+            IEdmStructuredType implicitRangeVariableType = null;
+
+            if (this.state != null)
+            {
+                implicitRangeVariableType = (IEdmStructuredType)this.state.ImplicitRangeVariable.TypeReference.Definition;
+            }
+
+            // When we have a derived type, the current edmType is different from the edmType of the RangeVariable type reference.
+            // e.g /Orders?$expand=Customer/Model.VipCustomer($select=VipCustomerName)
+            // We need to use the edmType of the RangeVariable type reference so that we can get the VipCustomerName property
+            // from the Model.VipCustomer.
+            if (implicitRangeVariableType != null && implicitRangeVariableType != this.edmType)
+            {
+                // Validate that the derived type and the base type are related.
+                UriEdmHelpers.CheckRelatedTo(this.state.ImplicitRangeVariable.TypeReference.Definition, this.edmType);

```

Comment:
The order of the arguments doesn't matter

Reply:
Not a blocker, but is there a test that confirms that by any chance?

Reply:
I have refactored that logic. We don't need to run that validation here since it's validated in the previous level binding i.e 
In this example
`Orders?$expand=Customer/NS.VipCustomer($select=VipCustomerName)`
We are binding the `$select`. However validation has already happened in the previous level `$expand=Customer/NS.VipCustomer`
https://github.com/OData/odata.net/blob/a510d92aba9ea220ae6de67e2ef659cd1d971b02/src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs#L392

Test are here https://github.com/OData/odata.net/blob/a510d92aba9ea220ae6de67e2ef659cd1d971b02/test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/SelectExpandFunctionalTests.cs#L1802